sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReactorToAzureContextWrapper.java

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation;
+
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * This class wraps a reactor context in the Azure-Core Context API, to avoid unnecessary copying
+ */
+public class ReactorToAzureContextWrapper extends Context {
+    private static final ClientLogger LOGGER = new ClientLogger(ReactorToAzureContextWrapper.class);
+
+    private final Map<String, String> contextAttributes;
+    private final reactor.util.context.ContextView reactorContext;
+    private final int size;
+
+    /**
+     * A wrapper class on top on reactor context, represent it as Azure Context type.
+     *
+     * @param reactorContext A reactor context object.
+     */
+    public ReactorToAzureContextWrapper(reactor.util.context.ContextView reactorContext) {
+        this(null, reactorContext);
+    }
+
+    /**
+     * A wrapper class on top on reactor context with additional attributes, represent it as Azure Context type.
+     *
+     * @param contextAttributes The attributes of context.
+     * @param reactorContext A reactor context object.
+     */
+    public ReactorToAzureContextWrapper(Map<String, String> contextAttributes,
+        reactor.util.context.ContextView reactorContext) {
+        super();
+        this.contextAttributes = contextAttributes;
+        this.reactorContext = reactorContext;
+        this.size = reactorContext.size() + (contextAttributes == null ? 0 : contextAttributes.size());
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public Optional<Object> getData(Object key) {
+        if (key == null) {
+            // going into the exception case
+            return super.getData(null);
+        }
+
+        // Look through the reactor context first
+        Optional<Object> value = reactorContext.getOrEmpty(key);
+        if (value.isPresent()) {
+            return value;
+        }
+
+        // then look through the map that we were given
+        if (contextAttributes != null && contextAttributes.containsKey(key)) {
+            return Optional.ofNullable(contextAttributes.get(key));
+        }
+
+        // otherwise defer up the context chain
+        return super.getData(key);
+    }
+
+    @Override
+    public Map<Object, Object> getValues() {
+        return new AbstractMap<Object, Object>() {
```

Comment:
Currently, the getValue() in Context return a HashMap but here we switch to more base class, AbstractMap. Does it consider a breaking change?  HashMap implements Map and extends AbstractMap @JonathanGiles

Reply:
Our API only states it returns Map, so we are free to change the implementation as long as the functionality remains intact.